### PR TITLE
Manual Backport of [HCP Telemetry] Move first TelemetryConfig Fetch into the TelemetryConfigProvider into release/1.15.x

### PR DESCRIPTION
**Pull request by Achooo**
Backport #18318 into `release/1.15.x`

The original backport PR had failed so I cherry picked the commit from main `0f48b7af5ec5fdf2237ba9e79a63160a071f6a91` here.

https://github.com/hashicorp/consul/pull/18619

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Comment by Achooo** [comment]
Not sure how to redeploy that Vercel failure, can't seem to login/ have access to hit the redeploy button.
Screenshot 2023-08-31 at 12 02 11 PM

**Comment by Achooo** [comment]
Oh I think its supposed to get skipped 🤔 This is from a different PR that got back ported recently https://github.com/hashicorp/consul/pull/18599

Screenshot 2023-08-31 at 12 09 34 PM

File: test-integ/go.mod [hunk 1]
```diff
@@ -0,0 +1,246 @@
+module github.com/hashicorp/consul/test-integ
```
Comment (chapmanc) [inline test-integ/go.mod:1]: What is this test-integ module for ? Also don't love a hypen in a package name https://go.dev/blog/package-names 

Finally this is  under consul which means it's a integration test for all of consul which I don't think you were intending to make?  Let me know what you think.

Comment (Achooo) [reply test-integ/go.mod:1]: Oh thanks - I need to remove this. I just cherry-picked the original commit. It's a bit weird but `main` has these files, but not this release branch.

I will remove these

**Review by chapmanc — APPROVED** [review]

**Pull request merged**